osu.Game/Screens/OnlinePlay/OnlinePlayScreen.cs

```diff
@@ -184,7 +180,7 @@ protected override void LoadComplete()
             screenStack.ScreenPushed += screenPushed;
             screenStack.ScreenExited += screenExited;
 
-            screenStack.Push(loungeSubScreen);
+            screenStack.Push(loungeSubScreen = CreateLounge());
```

Comment:
Pretty sure this delayed push is causing the test failures. Long and short of it is that it prevents guards on `multiplayerScreen.IsLoaded` on screens from working properly, since the root screen may be loaded, but its subscreen (which may handle input, etc.) is not necessarily so at that point.

Unsure whether the way forward is to fix the test or change something here - I understand that this move to `LoadComplete()` is integral to making this pull work?

Reply:
I'm not too sure yet, going to need some further investigation. I think maybe a change to the `ScreenStack` API may be required to make this work well.

Reply:
Seems like adding a load step is ample here. All the other flows which rely on the lounge to be loaded are guarded with `Schedule` calls in this PR.